lib/user.cpp

```diff
@@ -424,3 +427,15 @@ void User::processEvent(const RoomMemberEvent& event, const Room* room)
         updateAvatarUrl(event.avatarUrl(), d->avatarUrlForRoom(room), room);
     }
 }
+
+qreal User::hueF() const
+{
+    QByteArray hash = QCryptographicHash::hash(d->userId.toUtf8(),
+                                               QCryptographicHash::Sha1);
```

Comment:
`QCryptographicHash` is probably a bit of overkill here; a simple (and much quicker) `qHash()` should be enough. After all, as discussed in the room (long before - took me time to get to this PR...), we're not trying to reliably protect users from impersonation in that way (and SHA1 is cryptographically weak anyway).

Reply:
I've switched to SHA1 accordting to [xep-0392 logic](https://xmpp.org/extensions/xep-0392.html#design-mixing). About `qHash()`: in that case we should standardize some dictionary, right? :thinking:

Reply:
Oh; I totally glossed over this. Ok, let's keep it that way then. Then it would be probably good that this calculation occur just once per user and saved in `User::Private`.